ehr_billing/src/org/labkey/ehr_billing/EHR_BillingManager.java

```diff
@@ -52,43 +59,57 @@ public static EHR_BillingManager get()
     {
-        TableInfo invoiceRuns = EHR_BillingSchema.getInstance().getSchema().getTable(EHR_BillingSchema.TABLE_INVOICE_RUNS);
-        TableInfo invoicedItems = EHR_BillingSchema.getInstance().getSchema().getTable(EHR_BillingSchema.TABLE_INVOICED_ITEMS);
+        TableInfo invoice = QueryService.get().getUserSchema(user, container,EHR_BillingSchema.NAME).getTable(EHR_BillingSchema.TABLE_INVOICE);
+        TableInfo invoiceRuns = QueryService.get().getUserSchema(user, container,EHR_BillingSchema.NAME).getTable(EHR_BillingSchema.TABLE_INVOICE_RUNS);
+        TableInfo invoicedItems = QueryService.get().getUserSchema(user, container,EHR_BillingSchema.NAME).getTable(EHR_BillingSchema.TABLE_INVOICED_ITEMS);
         TableInfo miscCharges = EHR_BillingSchema.getInstance().getSchema().getTable(EHR_BillingSchema.TABLE_MISC_CHARGES);
 
         //create filters
-        SimpleFilter invoiceRunFilter = new SimpleFilter(FieldKey.fromString("invoiceId"), pks, CompareType.IN);
+        SimpleFilter objectIdFilter = new SimpleFilter(FieldKey.fromString("objectid"), pks, CompareType.IN);
+        SimpleFilter invoiceIdFilter = new SimpleFilter(FieldKey.fromString("invoiceId"), pks, CompareType.IN);
+        SimpleFilter invoiceRunIdFilter = new SimpleFilter(FieldKey.fromString("invoiceRunId"), pks, CompareType.IN);
 
         SimpleFilter miscChargesFilter = new SimpleFilter(FieldKey.fromString("invoiceId"), pks, CompareType.IN);
+        TableSelector tsInvItems = new TableSelector(invoicedItems, invoiceIdFilter, null);
+        TableSelector tsInvoice = new TableSelector(invoice, invoiceRunIdFilter, null);
 
         //perform the work
         List<String> ret = new ArrayList<>();
         if (testOnly)
         {
-            TableSelector tsRuns = new TableSelector(invoicedItems, invoiceRunFilter, null);
-            ret.add(tsRuns.getRowCount() + " records from invoiced items");
+            ret.add(tsInvItems.getRowCount() + " records from invoiced items");
+            ret.add(tsInvoice.getRowCount() + " records from invoice");
 
             TableSelector tsMiscCharges2 = new TableSelector(miscCharges, miscChargesFilter, null);
-            ret.add(tsMiscCharges2.getRowCount() + " records from misc charges will be removed from the deleted invoice, which means they will be picked up by the next billing period.  They are not deleted.");
+            ret.add(tsMiscCharges2.getRowCount() + " invoice records from misc charges will be removed from the deleted invoice, which means they will be picked up by the next billing period.  They are not deleted.");
         }
         else
         {
             try (DbScope.Transaction transaction = ExperimentService.get().ensureTransaction())
             {
-                long deleted1 = Table.delete(invoicedItems, invoiceRunFilter);
+                Map<String, Object>[] invoicedItemsRows = tsInvItems.getMapArray();
+                deleteInvoicedRows(invoicedItems, invoicedItemsRows, user, container);
+
```

Comment:
As discussed, recommend just doing an audit entry for the whole run's deletion, don't need separate audit entries for every invoicedItem row

Reply:
Done.
I've set audit logging to none for invoicedItems and invoice tables.
Also, briefly looked to see if I could change the audit log message to include no. of rows deleted from invoicedItems and invoice - but didn't find anything that was easy to do without creating a custom audit event and provider, unless you have suggestions if we want to go that route.

Reply:
This is now updated to use Table.delete()